### ISSUE TITLE
ObservableArray.reduce improvements

### ISSUE THREAD

**Issue by samreid** [phetsims/axon#203]
From #117 we determined that ObservableArray.reduce has the following problems:

1. It has a different signature than Array.reduce (args in different order)
2. It doesn't use the lodash or JS implementation
3. It is rarely called (maybe only once or twice).

We should delete it or fix it.

**Comment by samreid** [comment]
I will not have time for this in the near future, unassigning for now.

**Comment by samreid** [comment]
We expect to deprecate ObservableArray and move forward with AxonArray, see https://github.com/phetsims/axon/issues/314.  Putting this issue on hold for now.

**Comment by samreid** [comment]
ObservableArray has been deleted, and `createObservableArray` supports the native `reduce`, closing.
